ops/charm.py

```diff
@@ -880,16 +1006,16 @@ class RelationMeta:
 
     VALID_SCOPES = ['global', 'container']
 
-    def __init__(self, role: RelationRole, relation_name: str, raw: dict):
-        if not isinstance(role, RelationRole):
+    def __init__(self, role: RelationRole, relation_name: str, raw: '_RelationMetaDict'):
+        if not isinstance(role, RelationRole):  # pyright: reportUnnecessaryIsInstance=false
             raise TypeError("role should be a Role, not {!r}".format(role))
         self._default_scope = self.VALID_SCOPES[0]
         self.role = role
         self.relation_name = relation_name
         self.interface_name = raw['interface']
 
         self.limit = raw.get('limit')
-        if self.limit and not isinstance(self.limit, int):
+        if self.limit and not isinstance(self.limit, int):  # pyright: reportUnnecessaryIsInstance=false  # noqa
```

Comment:
same comment as above - maybe this check becomes unnecessary.

Reply:
I'm a bit hesitant to get rid of existing guards. Maybe I can weaken them to assertions so they can be skipped by optimized production code, but still raise when developing/testing

Reply:
Actually this typeerror specifically is a user-facing one. If the user were to set a (say) string or float limit in metadata.yaml, they'd get a TypeError. Not something we should get rid of lightly.

Reply:
fair enough.